File: 150_Evaluate_Reverse_Polish_Notation.py
class Solution(object):
    def process(self,s1,s2,s3):
        if s3=="+":
            return int(s1)+int(s2)
        elif s3=="-":
            return int(s1)-int(s2)
        elif s3=="*":
            return int(s1)*int(s2)
        elif s3=="/":
            return int(float(s1)/float(s2))
    
    def evalRPN(self, tokens):
        """
        :type tokens: List[str]
        :rtype: int
        """
        
        tokenStack = []
        for token in tokens:
            tokenStack.append(token)
            if token=="+" or token=="-" or token=="*" or token=="/":
                rtn=self.process(tokenQueue[-3],tokenQueue[-2],tokenQueue[-1])
                tokenStack.pop()
                tokenStack.pop()
                tokenStack.pop()
                tokenStack.append(rtn)
                
        return int(tokenStack[0])

File: 150_Evaluate_Reverse_Polish_Notation.cpp
#include<stack>
#include<string>
#include<iostream>

class Solution {
public:
    int evalRPN(vector<string>& tokens) {
        
        if(tokens.empty())
            return 0;
        
        int num1, num2, num;
        string item;
        std::stack<string> stk;
        for(int i=0; i<tokens.size(); i++){
            item = tokens[i];
            if(item=="+"){
                num2 = std::stoi(stk.top());
                stk.pop();
                num1 = std::stoi(stk.top());
                stk.pop();
                num = num1+num2;
                stk.push(std::to_string(num));
            }
            else if(item=="-"){
                num2 = std::stoi(stk.top());
                stk.pop();
                num1 = std::stoi(stk.top());
                stk.pop();
                num = num1-num2;
                stk.push(std::to_string(num));
            }
            else if(item=="*"){
                num2 = std::stoi(stk.top());
                stk.pop();
                num1 = std::stoi(stk.top());
                stk.pop();
                num = num1*num2;
                stk.push(std::to_string(num));
            }
            else if(item=="/"){
                num2 = std::stoi(stk.top());
                stk.pop();
                num1 = std::stoi(stk.top());
                stk.pop();
                num = num1/num2;
                stk.push(std::to_string(num));
            }
            else
                stk.push(item);
        }
        
        return std::stoi(stk.top());
        
    }
};
